primitives/src/vid/advz/payload_prover.rs

```diff
@@ -23,11 +24,37 @@ use super::{
 };
 use crate::{
     alloc::string::ToString,
-    vid::{payload_prover::PayloadProver, vid, VidError, VidScheme},
+    vid::{
+        payload_prover::{PayloadProver, Statement},
+        vid, VidError, VidScheme,
+    },
 };
 use ark_std::{format, ops::Range};
 
-impl<P, T, H, V> PayloadProver<Proof<P::Proof>> for GenericAdvz<P, T, H, V>
+/// A proof intended for use on small payload subslices.
+///
+/// KZG batch proofs and accompanying metadata.
+///
+/// TODO use batch proof instead of `Vec<P>` <https://github.com/EspressoSystems/jellyfish/issues/387>
+pub struct SmallRangeProof<P> {
```

Comment:
The proof types should at least derive `Clone`, `Debug`, and serde traits

Reply:
Definitely true. I'm going to merge now anyway because (i) this PR is already too big, (ii) don't want to initiate another round of approval, (iii) there will certainly be back-and-forth over issues like this as we integrate into the sequencer.